### DISCUSSION TITLE
`workerd-cxx` vs `nix-c-raw`

### DISCUSSION THREAD

**Discussion by srid** [nixops4/nixops4#44] [Q&A] [answered]
I just came across https://github.com/cloudflare/workerd-cxx

I'm curious whether this project has considered it before and if so why it went with [`bindgen` based `nix-c-raw`](https://github.com/nixops4/nixops4/tree/main/rust/nix-c-raw).

**Comment by scottbot95** [comment]
That `workerd-cxx` sounds pretty cool. My guess why it was decided to use bindgen is it has something to do with [this issue](https://github.com/NixOS/nix/issues/7271). Specifically the statement that "the C++ api is not stable". Although I'm not sure if that was more just referring to C++ itself not being stable (which may be handled automatically be `workerd-css`) or if the actual Nix C++ API itself isn't considered stable (in which case I think it is probably better to target the C bindings which were [_created_ to be a stable interface](https://github.com/NixOS/nix/pull/8699)).

I too would be curious to get @roberth's insight on this though as it seems like much of the activity on this project is currently around the C bindings.

**Comment by roberth** [comment, marked as answer]
As @scottbot95 notes, the Nix C++ interfaces are considered internal, and they can change arbitrarily. That makes a C++-based solution unsuitable as a dependency of this project. It might be an interesting solution for integrating Rust code into Nix itself, perhaps, although I don't see much value in that right now.
